_search-plugins/caching/field-data-cache.md

```diff
@@ -0,0 +1,67 @@
+---
+layout: default
+title: Field data cache
+parent: Caching
+grand_parent: Improving search performance
+nav_order: 20
+---
+
+# Field data cache
+
+The field data cache is a node-level in-memory cache which stores field data and global ordinals. These are data structures that enable aggregations and sorting on analyzed text fields. Since they can be expensive to compute, they are cached after creation in case they need to be reused.
+
+## What is field data?
+
+Field data is a data structure that enables quick sorting and aggregation on text fields. Text fields are stored in an inverted index, which is optimized for quick searching for specific terms, not iterating through individual documents. As this is necessary for aggregations or sorting, OpenSearch must process the inverted index into another data structure which is optimized for individual document access. This is field data.
+
+## What are global ordinals? 
+
+When computing aggregations on a given field, each Lucene segment assigns a unique ordinal to each unique term, enabling things like bucket aggregations. When computing an aggregation across the whole cluster, it's necessary to create a global ordinal for each globally unique term, and to store a mapping between these global ordinals and the segment ordinals. This enables combining the aggregation results from each segment. This mapping is then stored in the field data cache for reuse in future aggregations. 
+
+If the field has the mapping parameter `"eager_global_ordinals": true`, which defaults to false, global ordinals are computed and stored on index refresh rather than lazily at query time.
+
+For more information on global ordinals, see [Eager global ordinals]({{site.url}}{{site.baseurl}}/field-types/mapping-parameters/eager-global-ordinals).
+
+## Best practices
+
+By default, field data is disabled on text fields, because it can consume large amounts of heap memory. If you want to sort or aggregate on text fields, we recommend instead creating a keyword subfield. Keyword fields are not analyzed and therefore do not get stored in an inverted index or require building a separate data structure for aggregations.
```

Comment:
```suggestion
By default, field data is disabled on text fields because it can consume large amounts of heap memory. If you want to sort or aggregate on text fields, we recommend instead creating a keyword subfield. Keyword fields are not analyzed and therefore do not get stored in an inverted index or require building a separate data structure for aggregations.
```